sql/item_func.h

```diff
@@ -2677,30 +2671,49 @@ class Item_func_get_lock :public Item_long_func
   {
     return mark_unsupported_function(func_name(), "()", arg, VCOL_IMPOSSIBLE);
   }
+};
+
+
+class Item_func_get_lock :public Item_func_lock
+{
+  bool check_arguments() const
+  {
+    return args[0]->check_type_general_purpose_string(func_name()) ||
+           args[1]->check_type_can_return_real(func_name());
+  }
+  String value;
+ public:
+  Item_func_get_lock(THD *thd, Item *a, Item *b) :Item_func_lock(thd, a, b) {}
+  longlong val_int();
+  const char *func_name() const { return "get_lock"; }
+  bool fix_length_and_dec() { max_length=1; maybe_null=1; return FALSE; }
   Item *get_copy(THD *thd)
   { return get_item_copy<Item_func_get_lock>(thd, this); }
 };
 
-class Item_func_release_lock :public Item_long_func
+
+class Item_func_release_all_locks :public Item_func_lock
+{
+public:
+  Item_func_release_all_locks(THD *thd): Item_func_lock(thd)
+  { unsigned_flag=1; }
+  longlong val_int() override;
```

Comment:
recurring: s/override/final

Reply:
Every `virtual` method in `final` class implicitly become `final` because you can't override them anymore, right?

So, `val_int()` is actually `final` here. But to see that I need to look above to see that class is actually `final`. Instead, I could get the same information if method would be `final` instead. So, same semantics, but easy readability. Do you agree, @Daniel-Solo ?

Reply:
In general `final` is preferable because it has stronger semantics that `override` and allow more optimization, e.g. devirtualization.

Reply:
@kevgs agree